src/ImageGallery.jsx

```diff
@@ -652,16 +654,17 @@ export default class ImageGallery extends React.Component {
     const CENTER = 'center';
     const RIGHT = 'right';
 
-    switch (index) {
-      case (currentIndex - 1):
-        alignment = ` ${LEFT}`;
-        break;
-      case (currentIndex):
-        alignment = ` ${CENTER}`;
-        break;
-      case (currentIndex + 1):
-        alignment = ` ${RIGHT}`;
-        break;
+    if (index === currentIndex - 1) {
+      alignment = ` ${LEFT}`;
+    } else if (index === currentIndex) {
+      alignment = ` ${CENTER}`;
+    } else if (index === currentIndex + 1) {
+      alignment = ` ${RIGHT}`;
+    } else if (this.props.numberOfImagesToPreLoad > 0) {
+      const preLoadIndex = index - currentIndex;
+      if (preLoadIndex > 0 && preLoadIndex <= this.props.numberOfImagesToPreLoad) {
+        alignment = ` ${RIGHT}${preLoadIndex}`;
```

Comment:
so this method exists to produce a special classname for a piece of the gallery? could you explain to me how this supports preloading?

Reply:
Please see updated PR description.